test/api/encoding.js

```diff
@@ -12,19 +12,24 @@ function fixturePath(fixture) {
 }
 
 function readFixture(fixture) {
-  return fs.readFile(fixturePath(fixture));
+  return fs.promises.readFile(fixturePath(fixture));
 }
 
 const factories = {
   Buffer: fixture => readFixture(fixture),
-  Uint8Array: fixture => readFixture(fixture).then(buffer => Uint8Array.from(buffer)),
-  ArrayBuffer: fixture => readFixture(fixture).then(buffer => buffer.buffer),
-  DataView: fixture => readFixture(fixture).then(buffer => new DataView(buffer.buffer)),
+  Uint8Array: fixture => readFixture(fixture).then(buffer =>
+    new Uint8Array(buffer, buffer.byteOffset, buffer.length)),
```

Comment:
```suggestion
    new Uint8Array(buffer.buffer, buffer.byteOffset, buffer.byteLength)),
```

Also as a nit: could you change all the `buffer.length`s to `buffer.byteLength`s for symmetry with `byteOffset`?

Reply:
**Node**’s `Buffer` documentation says to use `buffer.length`: <https://nodejs.org/api/buffer.html#buffer_buf_byteoffset>.

Reply:
They are the same thing for `Buffer` since it's really a `Uint8Array`, but I think stylistically `byteLength` fits better.